clar2wasm/src/words/blockinfo.rs

```diff
@@ -316,6 +466,177 @@ mod tests {
+            };
+            let block_time = match block_time_val {
+                Value::UInt(val) => val,
+                _ => panic!("expected uint"),
+            };
+            let now = chrono::Utc::now().timestamp() as u128;
+
+            // The block time should be close to the current time, so let's give it
+            // a 10 second window, to be safe.
+            assert!(block_time >= now - 10);
+        }
+
+        #[test]
+        fn get_stacks_block_info_header_hash() {
+            let mut env = TestEnvironment::new(StacksEpochId::Epoch30, ClarityVersion::Clarity3);
+            env.advance_chain_tip(1);
+            let result = env
+                .evaluate("(get-stacks-block-info? header-hash u0)")
+                .expect("Failed to init contract.");
+            assert_eq!(
+                result,
+                Some(Value::some(Value::buff_from([0; 32].to_vec()).unwrap()).unwrap())
+            );
+        }
+
+        #[test]
+        fn get_tenure_info_time() {
+            let mut env = TestEnvironment::new(StacksEpochId::Epoch30, ClarityVersion::Clarity3);
+            env.advance_chain_tip(1);
+            let result = env
+                .evaluate("(get-tenure-info? time u0)")
+                .expect("Failed to init contract.");
+            let block_time_val = match result {
+                Some(Value::Optional(OptionalData { data: Some(data) })) => *data,
+                _ => panic!("expected value"),
+            };
+            let block_time = match block_time_val {
+                Value::UInt(val) => val,
+                _ => panic!("expected uint"),
+            };
+            let now = chrono::Utc::now().timestamp() as u128;
+
+            // The block time should be close to the current time, so let's give it
+            // a 10 second window, to be safe.
+            assert!(block_time >= now - 10);
+        }
+
+        #[test]
+        fn get_tenure_info_header_hash() {
+            let mut env = TestEnvironment::new(StacksEpochId::Epoch30, ClarityVersion::Clarity3);
+            env.advance_chain_tip(1);
+            let result = env
+                .evaluate("(get-tenure-info? burnchain-header-hash u0)")
+                .expect("Failed to init contract.");
+            assert_eq!(
+                result,
+                Some(Value::some(Value::buff_from([0; 32].to_vec()).unwrap()).unwrap())
+            );
+        }
+
+        #[test]
+        fn get_tenure_info_miner_address() {
+            let mut env = TestEnvironment::new(StacksEpochId::Epoch30, ClarityVersion::Clarity3);
+            env.advance_chain_tip(1);
+            let result = env
+                .evaluate("(get-tenure-info? miner-address u0)")
+                .expect("Failed to init contract.");
+            assert_eq!(
+                result,
+                Some(
+                    Value::some(Value::Principal(
+                        PrincipalData::parse("ST000000000000000000002AMW42H").unwrap()
```

Comment:
Where does this principal come from?

Reply:
Same for this, they use the same `get_miner_address` function.